File: C1_d2_hello/app/src/main/java/com/myapplication/hello_wrld/MainActivity.java
package com.myapplication.hello_wrld;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView mHelloTextView;
    private EditText mSomethingEditText;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mHelloTextView = findViewById(R.id.textView);
        mSomethingEditText = findViewById(R.id.editText);
    }

    public void onClick(View view) {

        if (mSomethingEditText.getText().length() == 0) {
            mHelloTextView.setText("Hello Everybody!");
        } else {
            mHelloTextView.setText("Hello " + mSomethingEditText.getText() + "!" );
        }

    }
}

File: Github_client/app/src/main/java/com/kts/githubclient/ui/activity/MainActivity.kt
package com.kts.githubclient.ui.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.kts.githubclient.R
import com.kts.githubclient.mvp.presenter.MainPresenter
import com.kts.githubclient.mvp.view.MainView
import com.kts.githubclient.ui.App
import com.kts.githubclient.ui.BackButtonListener
import com.kts.githubclient.ui.adapter.RepositoriesRVAdapter
import moxy.MvpAppCompatActivity
import moxy.presenter.InjectPresenter
import moxy.presenter.ProvidePresenter
import ru.terrakok.cicerone.android.support.SupportAppNavigator

class MainActivity : MvpAppCompatActivity(), MainView {

    val navigator = SupportAppNavigator(this, R.id.container)

    @InjectPresenter
    lateinit var presenter: MainPresenter

    var adapter: RepositoriesRVAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    @ProvidePresenter
    fun providePresenter() = MainPresenter(App.instance.getRouter())

    override fun init() {

    }

    override fun onResumeFragments() {
        super.onResumeFragments()
        App.instance.getNavigatorHolder().setNavigator(navigator)
    }

    override fun onPause() {
        super.onPause()
        App.instance.getNavigatorHolder().removeNavigator()
    }

    override fun onBackPressed() {
        supportFragmentManager.fragments.forEach {
            if(it is BackButtonListener && it.backClicked()){
                return
            }
        }
        presenter.backClicked()
    }


}
